### PR TITLE
Restructuring this repo.

### PR DESCRIPTION
**Pull request by Sfshaza**
This PR includes:

- Moved the xml and image files (from the last published version of the book) to an `archive` directory. Let's keep them around as reference.
- Under `code`, renamed `ch02` to `language-tour` and `ch03` to `library-tour`.
- Restructured all running examples to use the pub format, adding pubspec files, renaming directories and files to comply with our current standards.
- Moved all non-running examples into `language-tour/reference` or `library-tour/reference` subdirectories. (Some are partial examples and some are intentionally broken, for illustrative purposes.)
- Added an `.analysis_options` file to the `code` directory. This file is inherited by all examples under that directory.
- Updated the `README` and the `.gitignore` files.
- I also have a commit ready to go in the www repo which updates the language-tour and library-tour files which cross-reference these examples.

Opening the `code` directory in IntelliJ shows you any errors/warnings/hints from the analysis engine, all at once, in the "Dart Analysis" tool pane.  A subsequent PR will clean up linter complaints. (There are no strong mode errors at this time.)

@kwalrath, please review!

### CODE REVIEW

**Comment by Sfshaza** [comment]
- Renamed an `assert` example to `assert-example`. (It didn't like that the example was named after a reserved word.
- Added the `language-tour/reference` and  `library-tour/reference`

**Comment by Sfshaza** [comment]
- Renamed the `assert` example to `assert-example`. Unhappiness ensues when naming an example with a reserved keyword.
- Added `language-tour/reference` and `library-tour/reference` to the `.analysis-options` file to be excluded from static analysis.

Ready for your review, @kwalrath!

**Comment by Sfshaza** [comment]
Oh, I also added a few missing examples.

File: README.md [hunk 1]
```diff
@@ -14,8 +14,10 @@ Repo and testing info
 -----------
 The text in this repo is old; the latest text is in the
 dart-lang/www.dartlang.org repo under
```
Comment (kwalrath) [inline README.md]: www.dartlang.org -> site-www (right?)

File: code/.analysis_options [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Supported lint rules and documentation: http://dart-lang.github.io/linter/lints/
```
Comment (kwalrath) [inline code/.analysis_options]: Did you want to turn off some of these rules?

File: code/language-tour/list-map-default-function-parameters/bin/main.dart [hunk 3]
```diff
@@ -0,0 +1,19 @@
+doStuff({List<int> list: const[1, 2, 3],
+         Map<String, String> gifts: const{'first':  'paper',
+                                          'second': 'cotton',
+                                          'third':  'leather'}})
+{
```
Comment (kwalrath) [inline code/language-tour/list-map-default-function-parameters/bin/main.dart]: Shouldn't the `{` be on the previous line? (Did you run the formatter on this?)

File: code/ch02/ch02_meta/bin/ch02_proxy_or_implements.dart [hunk 4]
```diff
@@ -1,42 +0,0 @@
-class SomeClass {
```
Comment (kwalrath) [inline code/ch02/ch02_meta/bin/ch02_proxy_or_implements.dart]: Did you remove this because the code isn't used anywhere?

**Comment by kwalrath** [comment]
Aside from the few things I noted, LGTM.

File: code/ch02/ch02_meta/bin/ch02_proxy_or_implements.dart (hunk 4 above)
Comment (Sfshaza) [inline code/ch02/ch02_meta/bin/ch02_proxy_or_implements.dart]: Yes.

**Pull request merged**